scripts/clean-shrinkwrap.js

```diff
@@ -15,29 +15,222 @@
 
-console.log('Removing:', shrinkwrapIgnore.join(', '));
+console.log('Removing:', platformSpecificDependencies.join(', '));
 
 /**
- * Run an npm command
- * @param {Array} command - list of arguments
- * @returns {ChildProcess}
+ * @summary Get a shrinkwrap dependency object
+ * @function
+ * @private
+ *
+ * @param {Object} shrinkwrap - the shrinkwrap file contents
+ * @param {String[]} shrinkwrapPath - path to shrinkwrap dependency
+ * @returns {Object} shrinkwrap object
+ *
+ * @example
+ * const object = getShrinkwrapDependencyObject(require('./npm-shrinkwrap.json'), [
+ *   'drivelist',
+ *   'lodash'
+ * ]);
  */
-const npm = (command) => {
-  const npmBinary = os.platform() === 'win32' ? 'npm.cmd' : 'npm';
-  return spawn(npmBinary, command, {
-    cwd: path.join(__dirname, '..'),
-    env: process.env,
-    stdio: [ process.stdin, process.stdout, process.stderr ]
-  });
+const getShrinkwrapDependencyObject = (shrinkwrap, shrinkwrapPath) => {
+  return _.reduce(shrinkwrapPath, (accumulator, dependency) => {
+    return _.get(accumulator, [ 'dependencies', dependency ]);
+  }, shrinkwrap);
+};
+
+/**
+ * @summary Get a cleaned shrinkwrap dependency object
+ * @function
+ * @private
+ *
+ * @description
+ * This function transforms the output of `getShrinkwrapDependencyObject()` to
+ * omit unnecessary properties such as `from`, or `dependencies`.
+ *
+ * @param {Object} shrinkwrap - the shrinkwrap file contents
+ * @param {String[]} shrinkwrapPath - path to shrinkwrap dependency
+ * @returns {Object} shrinkwrap object
+ *
+ * @example
+ * const object = getPrettyShrinkwrapDependencyObject(require('npm-shrinkwrap.json'), [
+ *   'drivelist',
+ *   'lodash'
+ * ]);
+ */
+const getPrettyShrinkwrapDependencyObject = (shrinkwrap, shrinkwrapPath) => {
+  const object = getShrinkwrapDependencyObject(shrinkwrap, shrinkwrapPath);
+
+  if (_.isNil(object)) {
+    return null;
+  }
+
+  return {
+    name: _.last(shrinkwrapPath),
+    path: shrinkwrapPath,
+    version: object.version,
+    development: Boolean(object.dev),
+    optional: Boolean(object.optional)
+  };
 };
 
-npm([ 'rm', '--ignore-scripts' ].concat(shrinkwrapIgnore))
-  .once('close', () => {
-    console.log('Done.');
+/**
+ * @summary Get the manifest of a shrinkwrap dependency
+ * @function
+ * @private
+ *
+ * @param {String[]} shrinkwrapPath - path to shrinkwrap dependency
+ * @returns {Object} dependency manifest
+ *
+ * @example
+ * const manifest = getShrinkwrapDependencyManifest([ 'bluebird' ]);
+ */
+const getShrinkwrapDependencyManifest = (shrinkwrapPath) => {
+  const manifestPath = _.chain(shrinkwrapPath)
+    .flatMap((dependency) => {
+      return [ 'node_modules', dependency ];
+    })
+    .concat([ 'package.json' ])
+    .reduce((accumulator, file) => {
+      return path.join(accumulator, file);
+    }, '.')
+    .value();
+
+  try {
+    return require(`.${path.sep}${manifestPath}`);
+  } catch (error) {
+    if (error.code === 'MODULE_NOT_FOUND') {
```

Comment:
Hmmm, I've just seen in the NodeJS docs that `Using fs.access() to check for the accessibility of a file before calling fs.open(), fs.readFile() or fs.writeFile() is not recommended. Doing so introduces a race condition, since other processes may change the file's state between the two calls. Instead, user code should open/read/write the file directly and handle the error raised if the file is not accessible.`
Fair enough then ;-)